File: 0max-consecutive-ones/0max-consecutive-ones.go
func findMaxConsecutiveOnes(nums []int) int {
    cnt := 0
    k := 0
    for  _, n := range nums {
        if (n == 1) {
            cnt++
        } else {
            if k < cnt {
                k = cnt
            }
            cnt = 0
        }
	}
		
    if k > cnt {
        return k
    } else {
       return cnt
    }
}

File: 0max-consecutive-ones/0max-consecutive-ones.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int cnt = 0;
        int k = 0;
        for (int num : nums) {
            if (num == 0) {
                if (cnt < k) {
                    cnt = k;
                }
                k = 0;
            } else {
                k++;
            }
        }

        return Math.max(cnt, k);
    }
}